python/lsst/ts/standardscripts/auxtel/offset_ataos.py

```diff
@@ -0,0 +1,155 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["OffsetATAOS"]
+
+import yaml
+from lsst.ts.observatory.control.auxtel import ATCS, ATCSUsages
+
+from lsst.ts import salobj
+
+
+class OffsetATAOS(salobj.BaseScript):
+    """Perform an ATAOS offset.
+
+    Parameters
+    ----------
+    index : `int`
+        Index of Script SAL component.
+
+    Notes
+    -----
+
+    """
+
+    def __init__(self, index, add_remotes: bool = True):
+        super().__init__(
+            index=index,
+            descr="Perform an ATCS offset",
+        )
+
+        atcs_usage = None if add_remotes else ATCSUsages.DryTest
+
+        self.atcs = ATCS(domain=self.domain, intended_usage=atcs_usage, log=self.log)
+
+    @classmethod
+    def get_schema(cls):
+        schema_yaml = """
+            $schema: http://json-schema.org/draft-07/schema#
+            $id: https://github.com/lsst-ts/ts_standardscripts/auxtel/offset_atcs.yaml
+            title: OffsetATCS v1
+            description: Configuration for OffsetATCS Script.
+            type: object
+            properties:
+              z:
+                type: number
+                description: Offset hexapod in z axis (mm).
+              x:
+                type: number
+                description: Offset hexapod in x axis (mm).
+              y:
+                type: number
+                description: Offset hexapod in y axis (mm).
+              u:
+                type: number
+                description: Rx offset (deg).
+              v:
+                type: number
+                description: Ry offset (deg).
+              m1:
+                type: number
+                description: M1 pressure offset (Pa).
+                maximum: 0
+              offset_telescope:
+                type: bool
+                description: When correcting coma, also offset the telescope?
+                type: boolean
+                default: true
+              reset_offsets:
+                type: array
+                description: >-
+                    List of axes to reset. Must be one of z, x, y, z, u, v,
+                    m1, all
+                items:
+                    type: string
+
+            additionalProperties: false
+            anyOf:
+                - required: ["z"]
+                - required: ["x"]
+                - required: ["y"]
+                - required: ["u"]
+                - required: ["v"]
+                - required: ["m1"]
+                - required: ["reset_offsets"]
+        """
+        return yaml.safe_load(schema_yaml)
+
+    async def configure(self, config):
+        """Configure script.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+        """
+
+        self.offset_z = getattr(config, "z", 0.0)
```

Comment:
You can probably skip these assignments and go directly to the `self.offsets` dictionary in line 125. I suggest doing something like this:

```
self.offset = dict(**[getattr(config, axis, 0.0) for axis in ["x", "y", "z", "u", "v", "m1"])
```